tests/integration_tests/instances.py

```diff
@@ -43,21 +52,37 @@ def emit_settings_to_log(self) -> None:
     def destroy(self):
         self.instance.delete()
 
-    def execute(self, command):
-        return self.instance.execute(command)
+    def execute(self, command, *, use_sudo=None) -> Result:
+        if use_sudo is None:
+            use_sudo = self.use_sudo
+        return self.instance.execute(command, use_sudo=use_sudo)
 
-    def pull_file(self, remote_file, local_file):
-        self.instance.pull_file(remote_file, local_file)
+    def pull_file(self, remote_path, local_path):
+        # First copy to a temporary directory because of permissions issues
+        tmp_path = _get_tmp_path()
+        self.instance.execute('cp {} {}'.format(remote_path, tmp_path))
+        self.instance.pull_file(tmp_path, local_path)
 
     def push_file(self, local_path, remote_path):
-        self.instance.push_file(local_path, remote_path)
+        # First push to a temporary directory because of permissions issues
+        tmp_path = _get_tmp_path()
+        self.instance.push_file(local_path, tmp_path)
+        self.execute('mv {} {}'.format(tmp_path, remote_path))
 
     def read_from_file(self, remote_path) -> str:
-        tmp_file = NamedTemporaryFile('r')
-        self.pull_file(remote_path, tmp_file.name)
-        with tmp_file as f:
-            contents = f.read()
-        return contents
+        result = self.execute('/bin/cat {}'.format(remote_path))
```

Comment:
don't use full paths to things.

Reply:
I can't tell you number of times I've run into some stupid PATH alteration screwing something up because I didn't specify the full path. Admittedly, that's not likely to happen with `cat`, but I've always thought specifying full paths to things was the best practice. Why do you recommend not doing it?

Reply:
Because PATH exists solely for the purpose of allowing you to *not* specify full PATHs.  And if someone has taken the effort of updating PATH so that /mybin/program would be used before /bin/program, then they probably wanted that to happen.  Here is an example https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=852569.

And specifically with 'cat',  '/bin/cat' is now *actually* a legacy symlink to be pulled out from under you at some point in the future (https://wiki.debian.org/UsrMerge).  On a fresh focal container:

    root@f2:~# ls -l /bin
    lrwxrwxrwx 1 root root 7 Nov 17 21:40 /bin -> usr/bin
    root@f2:~# which cat
    /usr/bin/cat

Its really a very similar argument to not "fixing" the broken-ness of 'push_file' or 'pull_file' here, but rather fixing pycloudlib.  Stop fighting the system, and fix the system.  If PATH is wrong, then figure out why and fix it.